targets/disc-configs/lenovo-x1-carbon-disk-config.nix

```
@@ -0,0 +1,166 @@
+# Copyright 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+_: {
+  disko.devices = {
+    disk.disk1 = {
+      device = "DRIVE_PATH";
+      type = "disk";
+      content = {
+        type = "gpt";
+        partitions = {
+          boot = {
+            name = "boot";
+            size = "1M";
+            type = "EF02";
+          };
+          esp_a = {
+            name = "ESP_A";
+            size = "500M";
+            type = "EF00";
+            content = {
+              type = "filesystem";
+              format = "vfat";
+              mountpoint = "/boot";
+              mountOptions = ["umask=0077"];
```

I like adding `"nofail"` for non-root mounts. It makes recovering easier in case there is something wrong with mountpoints. Otherwise systemd will go into "emergency" mode. This can even happen on nixos-rebuild.

```suggestion
              mountOptions = ["umask=0077" "nofail"];
```